### DISCUSSION TITLE
Reduce bundle size

### DISCUSSION THREAD

**Discussion by garygreen** [inertiajs/inertia#758] [Ideas]
Currently the bundle is [59KB to parse:](https://bundlephobia.com/package/@inertiajs/inertia@0.9.2)

![image](https://user-images.githubusercontent.com/1702638/122594698-48654900-d05f-11eb-84eb-094bb92005ab.png)

In comparison, this is almost as large at the Vue runtime itself, and would be less with code splitting with v3 of Vue enabled

![image](https://user-images.githubusercontent.com/1702638/122594824-764a8d80-d05f-11eb-9b1b-2ee36ea2cc77.png)

It seems inertia is mostly made up of:

* axios - 25%
* (own code) - 19%
* qs library - 19%
* object-inspect - 13%
* get-intrinsic - 12%

Suggestions: 

axios - could fetch be used instead?
qs library - could `URLSearchParams` be used instead?
object-inspect - would assume it's heavily used in library, so no real alternative here
get-intrinsic - no idea what this is used for?

Is there a strong need for these libraries - in particular axios? Could fetch be used instead (even though it's not quite as nice API?)

**Comment by reinink** [comment]
Hey thanks for sharing these numbers. 👍

The plan is to eventually replace axios with XMLHttpRequest (see #216). We can't use fetch, since it doesn't support upload progress.
